system-int-test/test/cmr/system_int_test/search/generic_association_test.clj

```diff
@@ -45,6 +45,216 @@
 ;; Tests
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+;; Test that generic associations can be made between generic documents and tools.
+(deftest test-tool-and-generic-association
```

Comment:
For completeness, we should have tests for the not allowed associations to show an error message is returned.

Reply:
It's included in (deftest test-collection-and-generic-association